api/src/main/java/org/openmrs/module/insuranceclaims/api/service/fhir/util/InsuranceClaimUtil.java

```diff
@@ -0,0 +1,181 @@
+                ", " +
+                location.getTags();
+        locationReference.setDisplay(display);
+        locationReference.setId(location.getUuid());
+
+        return locationReference;
+    }
+
+    public static CodeableConcept createClaimVisitType(InsuranceClaim omrsClaim) {
+        String omrsVisitType = omrsClaim.getVisitType().getName();
+        CodeableConcept visitType = new CodeableConcept();
+        visitType.setText(omrsVisitType);
+        return visitType;
+    }
+
+    public static VisitType getClaimVisitType(Claim claim, List<String> errors) {
+        String visitTypeName = claim.getType().getText();
+        List<VisitType> visitType = getVisitTypeByName(visitTypeName);
+        if (visitType.size() > 1) {
+            errors.add("More than one matching visit type was found: %");
+            return null;
+        }
+        return visitType.get(0);
+    }
+
+    public static Patient getClaimPatient(Claim claim, List<String> errors) {
+        String patientUuid = FHIRUtils.getObjectUuidByReference(claim.getPatient());
+        Patient patient = Context.getPatientService().getPatientByUuid(patientUuid);
+        if (patient == null) {
+            errors.add("Could not find patient");
+            return null;
+        }
+        return patient;
+    }
+
+    public static Provider getClaimProvider(Claim claim, List<String> errors) {
+        String providerUUid = FHIRUtils.getObjectUuidByReference(claim.getProvider());
+        Provider provider = Context.getProviderService().getProviderByUuid(providerUUid);
+        if (provider == null) {
+            errors.add("Could not find provider");
+            return null;
+        }
+        return provider;
+    }
+
+    public static Location getClaimLocation(Claim claim, List<String> errors) {
+        String locationUuid = FHIRUtils.getObjectUuidByReference(claim.getFacility());
+        Location location = Context.getLocationService().getLocationByUuid(locationUuid);
+        if (location == null) {
+            errors.add("Could not find provider");
+            return null;
+        }
+        return location;
+    }
+
+    public static List<Identifier> createClaimIdentifier(InsuranceClaim omrsClaim) {
+        List<Identifier> fhirIdentifier = new LinkedList<>();
+        fhirIdentifier.add(createUuidIdentifier(omrsClaim));
+        fhirIdentifier.add(createClaimCodeIdentifier(omrsClaim));
+
+        return fhirIdentifier;
+    }
+
+    public static Map<String, Date> getClaimBillablePeriod(Claim claim, List<String> errors) {
+        Map<String, Date> claimPeriod = new HashMap<>();
+        Period billablePeriod = claim.getBillablePeriod();
+        Date from = billablePeriod.getStart();
+        if (from == null) {
+            errors.add("Date 'from' is missing");
+        }
+        Date to = billablePeriod.getEnd();
+        if (to == null) {
+            errors.add("Date 'to' is missing");
+        }
+        claimPeriod.put("from", from);
```

Comment:
"from" should be defined as constant and used also while getting information from this map